electronics_lib/Passives.py

```diff
@@ -314,6 +297,195 @@ def derated_capacitance(row: Dict[str, Any]) -> Tuple[float, float]:
+
+      for package in package_options:
+        if package.max >= min_nominal_capacitance[1]:
+            for package_max_voltage, package_max_capacitance in package.vc_pairs.items():
+              if package_max_voltage >= voltage[1] and package_max_capacitance >= min_nominal_capacitance[1]:
+                return package.name
+      return None
+
+    min_nominal_capacitance = (capacitance[0] / derating_coeff, capacitance[1] / derating_coeff)
+
+    if min_nominal_capacitance[1] > single_nominal_capacitance[1]:
+      num_caps = math.ceil(min_nominal_capacitance[0] / self.SINGLE_CAP_MAX)
+      assert num_caps * self.SINGLE_CAP_MAX < min_nominal_capacitance[1], "can't generate parallel caps within max capacitance limit"
+
+      self.assign(self.selected_derated_capacitance, (
+        num_caps * capacitance[0],
+        num_caps * capacitance[1],
+      ))
+
+      self.assign(self.selected_capacitance, (
+        num_caps * min_nominal_capacitance[0],
+        num_caps * min_nominal_capacitance[1],
+      ))
+
+      if footprint_spec == "":
+        split_package = 'Capacitor_SMD:C_1206_3216Metric'
+      else:
+        split_package = footprint_spec
+
+      cap_model = DummyCapacitor(capacitance=(self.SINGLE_CAP_MAX, self.SINGLE_CAP_MAX),
+                                      voltage=self.voltage, footprint_spec=split_package)
+      self.c = ElementDict[DummyCapacitor]()
+      for i in range(num_caps):
+        self.c[i] = self.Block(cap_model)
+        self.connect(self.c[i].pos, self.pos)
+        self.connect(self.c[i].neg, self.neg)
+    else:
+      valid_footprint_spec = select_package(min_nominal_capacitance, voltage)
+      assert valid_footprint_spec is not None, "cannot generate a valid footprint spec"
+      value = choose_preferred_number(min_nominal_capacitance, 0, self.E24_SERIES_ZIGZAG, 2)
+      assert value is not None, "cannot generate a preferred number"
+      self.assign(self.selected_derated_capacitance, capacitance)
+      self.assign(self.selected_capacitance, min_nominal_capacitance)
+
+      self.footprint(
+        'C', valid_footprint_spec,
+        {
+          '1': self.pos,
+          '2': self.neg,
+        },
+        value=f'{UnitUtils.num_to_prefix(value, 3)}F'
+      )
+
+class DummyCapacitor(DummyDevice, Capacitor, FootprintBlock, GeneratorBlock):
+  """
+  Capacitor that does not derate, used for splitting a generic capacitor into multiple when desired capacitance is too high
+  """
+
+  @init_in_parent
+  def __init__(self, footprint_spec: StringLike = "", *args, **kwargs):
+    super().__init__(*args, **kwargs)
+
+    self.footprint_spec = self.Parameter(StringExpr(footprint_spec))
+    self.generator(self.select_capacitor, self.capacitance, self.footprint_spec)
+
+    # Output values
+    self.selected_capacitance = self.Parameter(RangeExpr())
+    self.selected_voltage_rating = self.Parameter(RangeExpr())
```

Comment:
Since there's no selection in this class / generator, can these just be deleted? (or in other words, would this data provide the user with any new information / insight?)